torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -609,6 +608,10 @@ void Reducer::prepare_for_backward(
 
 // A bucket with one or more dense tensors needs to be unflattened.
 void Reducer::finalize_bucket_dense(Bucket& bucket) {
+  AT_ASSERTM(
+      !local_used_maps_reduced_,
+      "local_used_maps should have not been reduced yet at this point");
+
```

Comment:
This may trigger. This function is called once per bucket, so if the first one waits for the reduction and sets `local_used_maps_reduced_`, then the second bucket will trip this assert.

Also please use `TORCH_INTERNAL_ASSERT` instead of `AT_ASSERTM`. The latter is being deprecated.

Reply:
Ah, that's right. I'll remove the assert then.

Reply:
Done with bc1c493